### ISSUE TITLE
Block extensions from chording Control + R on Linux when terminal is focused

### ISSUE THREAD

**Issue by tristan957** [microsoft/vscode#103581]
Control + R in a terminal shell is generally a historical recursive search. _Recently_ (meaning for the past couple of versions maybe since ~1.40) Control + R in VSCode on Linux has, instead of showing recursive search in my terminal, been using Control + R as a chorded keyboard shortcut even when the terminal is in focus. This was not the case in the past.

I would like to propose that the terminal have higher precedence in the keyboard shortcut hierarchy when Control + R is pressed and terminal is focused. This issue doesn't affect my mac because all the Control + R keyboard shortcuts bind to Command + R instead.

I am unsure of whether to take this request to individual extensions like the ~C# and~ NPM extension~s~ from Microsoft, but it feels like a toplevel VSCode issue.

**Comment by tristan957** [comment]
And since I am a lazy dev, I have had to keep an external shell open so I could use Control + R.

**Comment by Tyriar** [comment]
> I am unsure of whether to take this request to individual extensions like the C# and NPM extensions from Microsoft, but it feels like a toplevel VSCode issue.

I don't see any keybindings on the C# or npm extensions, are you sure they're the ones to blame? Ca

I'm reluctant to add specific rules like this because why stop at R, when there are plenty of other ctrl+ keybindings used in the terminal, why stop at ctrl when alt is also used, etc.

**Comment by tristan957** [comment]
> I'm reluctant to add specific rules like this because why stop at R, when there are plenty of other ctrl+ keybindings used in the terminal, why stop at ctrl when alt is also used, etc.

That makes sense. I looked at the C3 extension again and it has good `when` conditions, and is not a part of the problem. I just tried disabling the NPM extension and suddenly Control + R could be used in the terminal again. I think your reasoning is good, and I will take this issue to the NPM extension directly.

The commands I am seeing that conflict with Control + R being provided by the NPM extension are

```
Rerun last script (ctrl+r, r)
Run script (ctrl+r, R)
Run test (ctrl+r, t)
Show output (ctrl+r, l)
Terminate script (ctrl+r, X)
```

Source: https://github.com/microsoft/vscode-npm-scripts/blob/ebe4f49e155cc5d71de10d1bd85d6306729dc4c7/package.json#L126
